java/cdk-environment/src/test/java/sleeper/environment/cdk/buildec2/KeyPairUtilTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.buildec2;
+
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KeyPairUtilTest {
+
+    @Test
+    public void canGetPublicKeyInBase64() {
+        assertThat(KeyPairUtil.publicBase64(KeyPairUtil.generate()))
+                .hasSize(392);
+    }
+
+    @Test
+    public void canGetPrivateKeyAsPemString() {
+        assertThat(KeyPairUtil.privatePem(KeyPairUtil.generate()))
+                .startsWith("-----BEGIN RSA PRIVATE KEY-----\n")
+                .endsWith("\n-----END RSA PRIVATE KEY-----\n")
+                .hasLineCount(28)
+                .hasSizeGreaterThan(1700);
+    }
+
+    @Test
+    public void canWritePrivateKeyFile() throws Exception {
+        Path expectedPath = Paths.get("test.pem");
```

Comment:
Could be worth putting the file name in a variable to ensure the same file is referenced in the test, and also making sure it's a different file for each test (to allow parallel unit testing)